File: modules/coupon-domain/src/main/java/com/thanh0x/coursedeal/dto/PreferenceDTO.java
package com.thanh0x.coursedeal.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

/**
 * DTO for user preferences.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PreferenceDTO {
    private Set<String> categories;
    private Set<String> keywords;
    private boolean notificationsEnabled;
}

File: modules/coupon-domain/src/main/java/com/thanh0x/coursedeal/dto/PreferenceDTO.kt
package com.thanh0x.coursedeal.dto

import jakarta.validation.constraints.NotNull

/**
 * DTO for user preferences.
 */
data class PreferenceDTO(
    @field:NotNull(message = "Categories set cannot be null")
    val categories: Set<String> = emptySet(),

    @field:NotNull(message = "Keywords set cannot be null")
    val keywords: Set<String> = emptySet(),

    val notificationsEnabled: Boolean = false
)
